crates/ethereum/evm/src/execute.rs

```diff
@@ -169,7 +172,8 @@ where
                     error: Box::new(new_err),
                 }
             })?;
-            evm.db_mut().commit(state);
+            evm.db_mut().commit(state.clone());
+            system_caller.on_state(&ResultAndState { result: result.clone(), state });
```

Comment:
these should be swapped, and we can delay the resultandstate destruct on L167, then we can get rid of the clone

Reply:
good catch, done